lib/components/inputs/FormCustomSelect.tsx

```diff
@@ -1,49 +1,24 @@
-import { FormInputProps } from "../../types";
-import { Controller, useFormContext } from "react-hook-form";
-
 import { CustomSelect } from "@atomicjolt/atomic-elements";
 import type { CustomSelectProps } from "@atomicjolt/atomic-elements";
+import { Controller } from "react-hook-form";
+import { FormInputProps } from "../../types";
+import { useControllerField } from "../../hooks/useControllerField";
 
-import { splitFormProps } from "../../hooks/useFormInput";
-import { errorMessage } from "../errors";
-
-export type FormCustomSelectProps<T extends object> = FormInputProps<
-  CustomSelectProps<T>,
-  | "valueAsDate"
-  | "valueAsNumber"
-  | "setValueAs"
-  | "max"
-  | "maxLength"
-  | "min"
-  | "minLength"
->;
+export interface FormCustomSelectProps<T extends object>
+  extends FormInputProps<
+    CustomSelectProps<T>,
+    CustomSelectProps<T>["selectedKey"]
+  > {}
 
 export function FormCustomSelect<T extends object>(
   props: FormCustomSelectProps<T>
 ) {
-  const [name, options, componentProps] = splitFormProps<
-    CustomSelectProps<T>,
-    FormInputProps<CustomSelectProps<T>>
-  >(props);
-
-  const { control } = useFormContext();
-
-  return (
-    <Controller
-      name={name}
-      rules={options}
-      control={control}
-      render={({ field: { onBlur, onChange, value }, fieldState }) => {
-        return (
-          <CustomSelect
-            onBlur={onBlur}
-            onChange={onChange}
-            value={value}
-            {...componentProps}
-            error={errorMessage(props.name, fieldState.error)}
-          />
-        );
-      }}
-    />
-  );
+  const controlProps = useControllerField(props, CustomSelect, {
+    aliases: {
+      value: "selectedKey",
+      defaultValue: "defaultSelectedKeys",
+      onChange: "onSelectionChange",
+    },
+  });
+  return <Controller {...controlProps} />;
```

Comment:
Not sure I understand how this works. It looks like  you were able to remove a lot of lines and just do a splat in its place?

Reply:
Yeah, essentially. 

All of the components use the `Controller` instead of most of them handling things via refs. Because of that, it simplified most of the implementation as they became consistent